### PR TITLE
Flake8 compliance --ignore=W503,E402,E721,E731

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This PR makes the table subpackage pass:
```
flake8 --ignore=W503,E402,E721,E731
```
This allows much better local flake8 testing during development with an IDE or command-line run with flake8.  It may eventually allow more complete CI testing using `--ignore` instead of `--select`, but that is not in scope here.

Justification for exceptions:

- W503: *Line break occurred before a binary operator.*  Best practice is now:
```
income = (gross_wages
          + taxable_interest)
```
- E402: *Module level import not at top of file.*  There are frequently good reasons for out-of-order imports, and the `# noqa` get distracting.
- E721: *Do not compare types, use 'isinstance()'.*  In testing it is frequently necessary to do direct type comparisons.  (There are around 40 instances of this in table).
- E731: *Do not assign a lambda expression, use a def.*  This one is controversial anyway, but IMO there are many cases where lamdba expressions provide for concise and readable code.

### CODE REVIEW

File: astropy/table/np_utils.py [hunk 1]
```diff
@@ -107,7 +103,7 @@ def get_descrs(arrays, col_name_map):
         # Make sure all input shapes are the same
         uniq_shapes = set(col.shape[1:] for col in in_cols)
         if len(uniq_shapes) != 1:
-            raise TableMergeError(f'Key columns {name!r} have different shape')
+            raise TableMergeError(f'Key columns have different shape')
```
Comment (taldcroft) [inline astropy/table/np_utils.py]: This is a real bug that showed up where an undefined variable was used in an exception.  Bad coverage...

**Comment by taldcroft** [comment]
@bsipocz @aarchiba - travis CI is failing on a cache problem.  Do I need #9703?
```
E               astropy.utils.data.CacheDamaged: Apparently abandoned files: {'/tmp/pytest-of-travis/pytest-0/test_check_download_cache0/astropy/download/py3/6fa27f095b6862050fdc465f0c29c850', '/tmp/pytest-of-travis/pytest-0/test_check_download_cache0/astropy/download/py3/eb145695fe8cc89a8ea7e5c74108d768'}
```

**Comment by pllim** [comment]
Ignore it for now. Temporarily skipping it in #9776. Still unresolved.

**Comment by taldcroft** [comment]
@mhvk - ping to have a look at this.

**Comment by taldcroft** [comment]
@mhvk - another ping to have a look at this please.  I'm looking at #9962 but get annoyed because my editor now is strict with flake8 so I see of a ton of false-positive errors in the table code.

**Pull request merged**